### PR TITLE
Askmethat rating

### PR DESCRIPTION
**Pull request by AlexTeixeira**
Update to fix the plugin issues, increase unit tests and better source code.

More information about the changes in this link : https://github.com/AlexTeixeira/Askmethat-Rating/releases/tag/0.3.3

### CODE REVIEW

File: types/askmethat-rating/index.d.ts [hunk 1]
```diff
@@ -75,7 +75,12 @@ export class AskmethatRating {
      *
      * @return  options based on @type AskmethatRatingOptions
      */
-    readonly defaultOptions: any;
+    /**
+     * @function set the default option for the rating
+     *
+     * @return  options based on @type AskmethatRatingOptions
+     */
+    defaultOptions: AskmethatRatingOptions;
```
Comment (andy-ms) [inline types/askmethat-rating/index.d.ts]: Is the change to mutable intentional?

Comment (AlexTeixeira) [inline types/askmethat-rating/index.d.ts]: Yes it is.  I had not check the **generated d.ts**.

**Comment by andy-ms** [comment]
Thanks!

**Comment by andy-ms** [comment]
Since you're the library author, have you considered [bundling](http://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html) the types yourself instead of using DefinitelyTyped?

**Comment by AlexTeixeira** [comment]
It's already done. 
But I added the types for people that have a preference to install types using npm (since Typescript 2 I don't use tsconfig anymore to manage types). 

And I hope one day somebody will use & modify my library ^^

**Comment by andy-ms** [comment]
> for people that have a preference to install types using npm

If your library is distributed on NPM, you should be able to include the types along with that package, so people who `npm install askmethat-rating` will get the types without having to also `npm install @types/askmethat-rating`.

**Pull request merged**